test/http2/secure-with-fallback.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+const fs = require('fs')
+const path = require('path')
+const Fastify = require('../..')
+const http2 = require('http2')
+const sget = require('simple-get').concat
+const msg = { hello: 'world' }
+
+var fastify
+try {
+  fastify = Fastify({
+    http2: true,
+    https: {
+      allowHTTP1: true,
```

Comment:
This option naming feels weird.

How about:
```js
http2: Boolean,
http1: Boolean
```
or:
```js
https: {
  allowHTTP1: Boolean,
  allowHTTP2: Boolean
},
http: {
  allowHTTP1: Boolean,
  allowHTTP2: Boolean
  // Caveat: Not yet possible to enable both simultaneously
}
```

Reply:
I don't like this either. The benefit of the current API is that it's transparent, as instead of going to `http`  and `https`  it just goes to `http2`. `allowHTTP1` option is from the `http2`  module in core.